src/H5FDsubfiling/H5FDioc.c

```diff
@@ -235,10 +235,18 @@ H5FD_ioc_init(void)
                 /* If MPI is initialized, validate that it was initialized with MPI_THREAD_MULTIPLE */
                 if (MPI_SUCCESS != (mpi_code = MPI_Query_thread(&provided)))
                     H5_SUBFILING_MPI_GOTO_ERROR(H5I_INVALID_HID, "MPI_Query_thread failed", mpi_code);
-                if (provided != MPI_THREAD_MULTIPLE)
+                if (provided != MPI_THREAD_MULTIPLE) {
+#ifdef MPICH
```

Comment:
We've tried to avoid these sorts of "platform" specific ifdef's in the library code.  This would be better a generic error message, with a pointer into the documentation, which could explain about the MPICH case.